cypress-io/cypress
# feat: display framework definition errors

- Closes #25838

### Additional details


This PR adds the ability for the launchpad wizard to surface warnings when third party CT framework definitions are **detected** (they follow the naming convention and are installed in the project), but are unable to be resolved or parsed correctly.

This will be useful for:
- Users who are creating and testing their own third party framework definition for use with Cypress
- Users who are using a third party definition and are having issues with it (this way, it will be obvious that something is wrong, and which framework definition is having the issue).

### Steps to test


- Install an invalid CT framework definition that still matches the package naming convention into a project that is not configured for CT (you can use `cypress-ct-bad-missing-value` or `cypress-ct-bad-syntax` that were added to the system tests projects in this PR)
- Launch Cypress, choose CT
- Verify that the warning alert is showing with a list of warnings, one for each of your invalid CT framework definitions that you have installed
- Verify that the text appropriately reflects the singular or plural form based on the number of definitions that were found
- Verify that links open in the browser.

### How has the user experience changed?


![Screenshot 2023-04-03 at 2 27 29 PM](https://user-images.githubusercontent.com/8340719/229644297-aa717a57-d58c-44dd-bbc5-3f1c4a549222.png)

### PR Tasks


- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
